src/OpenRiaServices.Server.EntityFrameworkCore/Framework/EFCoreTypeDescriptor.cs

```diff
@@ -0,0 +1,251 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace OpenRiaServices.Server.EntityFrameworkCore
+{
+    /// <summary>
+    /// CustomTypeDescriptor for LINQ To Entities
+    /// </summary>
+    internal class EFCoreTypeDescriptor : TypeDescriptorBase
+    {
+        private readonly EFCoreTypeDescriptionContext _typeDescriptionContext;
+        private readonly IEntityType _entityType;
+        private readonly IProperty _timestampProperty;
+        private readonly bool _keyIsEditable;
+        private Dictionary<string, IProperty> _foreignKeyMembers; // TODO: change to List<string> / HashSet<string>
+
+        public EFCoreTypeDescriptor(EFCoreTypeDescriptionContext typeDescriptionContext, IEntityType entityType, ICustomTypeDescriptor parent)
+        : base(parent)
+        {
+            _typeDescriptionContext = typeDescriptionContext;
+            _entityType = entityType;
+
+            var timestampMembers = entityType.GetProperties().Where(p => IsConcurrencyTimestamp(p)).ToArray();
+            if (timestampMembers.Length == 1)
+            {
+                _timestampProperty = timestampMembers[0];
+            }
+
+            // if (edmType.BuiltInTypeKind == BuiltInTypeKind.EntityType)
```

Comment:
should this be used? otherwise might want to remove { }

Reply:
There is also a TODO on foreignKeyMembers to change it to a List or HashSet

Reply:
I am ok with keeping this commented out code here (and maybe some other cases) as a reminder that there might be missing logic here, untill I know what other values of "edmType.BuiltInTypeKind" this check protects against for EF6.

It is very possible that the check is not needed and once it is understood why, it would be good to remove the code

Reply:
ok, for the checks in this file it seems like it excludes the case where `edmType.BuiltInTypeKind == BuiltInTypeKind.ComplexType`.

Reply:
it seems [owned entities](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities)  corresponds to ComplexTypes, I am fine with having an Issue for supporting "owned entities" or at least a TODO.md /README.md in the efcore project about this not yet beeing supported and some hints on what to change / where to look when supporing it.

I did however make a commit with some preliminary code for this, but adding final support for it would mean adding some units tests similar to `DomainServiceDescriptionTest.ComplexType_EFProviderTest` and related tests

Reply:
Documented under "Not supported / limited support"